src/fetch.rs

```diff
@@ -184,13 +186,29 @@ fn get_no_latest_version_from_json_when_all_are_yanked() {
     assert!(read_latest_version(&versions, false).is_err());
 }
 
-fn fetch_cratesio(path: &str) -> Result<Versions> {
-    let url = format!("{host}/api/v1{path}", host = REGISTRY_HOST, path = path);
-    let response =
-        get_with_timeout(&url, get_default_timeout()).chain_err(|| ErrorKind::FetchVersionFailure)?;
-    let versions: Versions =
-        json::from_reader(response).chain_err(|| ErrorKind::InvalidCratesIoJson)?;
-    Ok(versions)
+fn fetch_cratesio(crate_name: &str) -> Result<Versions> {
+    let url = format!(
+        "{host}/api/v1/crates/{crate_name}",
+        host = REGISTRY_HOST,
+        crate_name = crate_name
+    );
+
+    match get_with_timeout(&url, get_default_timeout()) {
+        Ok(response) => {
+            Ok(json::from_reader(response).chain_err(|| ErrorKind::InvalidCratesIoJson)?)
+        }
+        Err(e) => {
```

Comment:
We can use pattern guard here, I guess:
```rust
Err(e) if e.status() == Some(reqwest::StatusCode::NotFound) => { ...}
_ => { ... }
```

Reply:
Alas, `cannot bind by-move into a pattern guard`. I've not found a better way of doing this.